### PR TITLE
Fix menu on 'virtual solution items'

### PR DESCRIPTION
**Pull request by blackboxlogic**
Fixes https://github.com/idealist1508/ShellContextMenuExtension/issues/2
The context menu now works on files inside virtual solution folders. To get this running on my machine, I needed to make changes to project references.

### CODE REVIEW

**Comment by blackboxlogic** [comment]
I've added a change to fix https://github.com/idealist1508/ShellContextMenuExtension/issues/1

**Comment by idealist1508** [comment]
thanks for both PR's. I will look at soon and publish it.

**Comment by idealist1508** [comment]
It seems, that it didn't fix #2.
grafik

grafik

**Comment by idealist1508** [comment]
Fix for #1 is awesome.

File: VsShellContext/VsShellContextPackage.cs [hunk 1]
```diff
@@ -138,21 +138,23 @@ private IList<string> PopulateFileNamesFromSolutionExplorer()
             foreach (var hierarchyItem in selHierarchyItems)
             {
                 // Top-level project. See below for projects in folders.
-                var project = hierarchyItem.Object as Project;
-                if (project != null)
+                if (hierarchyItem.Object is Project project)
```
Comment (idealist1508) [inline VsShellContext/VsShellContextPackage.cs:141]: I think we shouldn't use c# 7 features in VS2015 project. (at least please add compiler package to solution)

Comment (blackboxlogic) [reply VsShellContext/VsShellContextPackage.cs]: Ah yes, I need to resist the urge to "improve" stuff. I'll put this back to what it was.

**Comment by blackboxlogic** [comment]
> It seems, that it didn't fix #2.
Oops! I had only tested a file which was also included in a project, and I think that only worked by accident.

I found [the answer](https://docs.microsoft.com/en-us/dotnet/api/envdte.projectitem.filenames?view=visualstudiosdk-2019#EnvDTE_ProjectItem_FileNames_System_Int16_) in a stack overflow [post](https://stackoverflow.com/a/3960162).

I want to make sure I'm testing all the scenarios but I am not able to explore the code path with the comment `// Projects inside a Solution Folder (i.e. "virtual" folder) gets wrapped inside another project item.` When I put a project inside a solution folder, it gets handled correctly by `// Top-level project. See below for projects in folders.` I don't dare delete this section though, assuming I've missed something while testing.

**Comment by idealist1508** [comment]
I just found that nanangp made an update to origin plugin  https://github.com/nanangp/vs-shell-contextmenu
It should be a better place for futere development. I will wait till nanangp publish your PR and then delete this repo.

**Comment by blackboxlogic** [comment]
Funny story, I actually had the other plugin installed (with the same bug) but found your repo by mistake and proceeding to fix (without noticing). Big old face-palm.
I wasn't going to mention it until the other PR got merged. Thanks for maintaining/hosting/publishing this as long as you did. Best of luck on your other adventures.

**Pull request merged**